File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/OnboardingModels.kt
package com.lextures.android.core.lms

import com.lextures.android.R
import kotlinx.serialization.Serializable

enum class PriorKnowledgeLevel(val wire: String) {
    Beginner("beginner"),
    Intermediate("intermediate"),
    Advanced("advanced"),
}

@Serializable
data class OnboardingStatus(
    val completed: Boolean = false,
    val step: Int = 0,
    val shouldShowFlow: Boolean = true,
)

@Serializable
data class LearnerGoals(
    val id: String = "",
    val userId: String = "",
    val topic: String = "",
    val goalText: String? = null,
    val targetDate: String? = null,
    val dailyMinutes: Int = 20,
    val priorKnowledgeLevel: String = PriorKnowledgeLevel.Beginner.wire,
    val diagnosticScore: Double? = null,
    val diagnosticSkipped: Boolean = false,
    val onboardingStep: Int = 0,
    val onboardingCompleted: Boolean = false,
    val reminderOptIn: Boolean = false,
    val reminderTime: String? = null,
    val recommendedCourseCode: String? = null,
    val recommendedCourseTitle: String? = null,
)

@Serializable
data class DiagnosticQuestion(
    val id: String,
    val prompt: String,
    val choices: List<String> = emptyList(),
)

@Serializable
data class GoalsEnvelope(
    val goals: LearnerGoals,
)

@Serializable
data class DiagnosticQuestionsResponse(
    val questions: List<DiagnosticQuestion> = emptyList(),
)

data class OnboardingTopic(
    val id: String,
    val labelRes: Int,
)

val onboardingTopics: List<OnboardingTopic> = listOf(
    OnboardingTopic("python", R.string.mobile_onboarding_topic_python),
    OnboardingTopic("javascript", R.string.mobile_onboarding_topic_javascript),
    OnboardingTopic("data-science", R.string.mobile_onboarding_topic_dataScience),
    OnboardingTopic("math", R.string.mobile_onboarding_topic_math),
    OnboardingTopic("business", R.string.mobile_onboarding_topic_business),
    OnboardingTopic("design", R.string.mobile_onboarding_topic_design),
)

enum class OnboardingStep(val value: Int) {
    Welcome(0),
    Topic(1),
    Experience(2),
    Diagnostic(3),
    Habits(4),
    Consent(5),
    Complete(6),
}

File: clients/ios/Lextures/Core/LMS/OnboardingModels.swift
import Foundation

enum PriorKnowledgeLevel: String, CaseIterable, Codable {
    case beginner
    case intermediate
    case advanced
}

struct OnboardingStatus: Decodable {
    var completed: Bool
    var step: Int
    var shouldShowFlow: Bool
}

struct LearnerGoals: Decodable {
    var id: String
    var userId: String
    var topic: String
    var goalText: String?
    var targetDate: String?
    var dailyMinutes: Int
    var priorKnowledgeLevel: PriorKnowledgeLevel
    var diagnosticScore: Double?
    var diagnosticSkipped: Bool
    var onboardingStep: Int
    var onboardingCompleted: Bool
    var reminderOptIn: Bool
    var reminderTime: String?
    var recommendedCourseCode: String?
    var recommendedCourseTitle: String?
}

struct DiagnosticQuestion: Decodable, Identifiable, Hashable {
    var id: String
    var prompt: String
    var choices: [String]
}

struct OnboardingTopic: Identifiable {
    var id: String
    var labelKey: String.LocalizationValue

    static let all: [OnboardingTopic] = [
        OnboardingTopic(id: "python", labelKey: "mobile.onboarding.topic.python"),
        OnboardingTopic(id: "javascript", labelKey: "mobile.onboarding.topic.javascript"),
        OnboardingTopic(id: "data-science", labelKey: "mobile.onboarding.topic.dataScience"),
        OnboardingTopic(id: "math", labelKey: "mobile.onboarding.topic.math"),
        OnboardingTopic(id: "business", labelKey: "mobile.onboarding.topic.business"),
        OnboardingTopic(id: "design", labelKey: "mobile.onboarding.topic.design"),
    ]
}

struct GoalsEnvelope: Decodable {
    var goals: LearnerGoals
}

struct DiagnosticQuestionsResponse: Decodable {
    var questions: [DiagnosticQuestion]
}

enum OnboardingStep: Int, CaseIterable {
    case welcome = 0
    case topic = 1
    case experience = 2
    case diagnostic = 3
    case habits = 4
    case consent = 5
    case complete = 6
}
